### PR TITLE
Use the new api_config in all services

### PR DESCRIPTION
**Pull request by gctucker**
Replace the now-deprecated db_config with the new api_config which is pending to be merged in kernelci-core with https://github.com/kernelci/kernelci-core/pull/1650

Update all the services accordingly.  This is to start transitioning away from the old Database abstraction layer class in kernelci-core into supporting only the KernelCI API.

### CODE REVIEW

**Comment by gctucker** [comment]
Tested OK on staging with `kver`:
```
02/10/2023 05:35:10 PM UTC [INFO] Listening for available checkout events
02/10/2023 05:35:10 PM UTC [INFO] Press Ctrl-C to stop.
Getting kernel source tree...
Source directory: /tmp/kci/linux-kernelci-staging-mainline-staging-mainline-20230210.8
Running job...
Checking kernel revision
Result: pass
02/13/2023 09:50:56 AM UTC [INFO] Listening for available checkout events
02/13/2023 09:50:56 AM UTC [INFO] Press Ctrl-C to stop.
Getting kernel source tree...
Source directory: /tmp/kci/linux-kernelci-staging-mainline-staging-mainline-20230213.0
Running job...
Checking kernel revision
Result: pass
Getting kernel source tree...
Source directory: /tmp/kci/linux-kernelci-staging-next-staging-next-20230212.0
Running job...
Checking kernel revision
Result: pass
Getting kernel source tree...
Source directory: /tmp/kci/linux-kernelci-staging-stable-staging-stable-20230213.0
Running job...
Checking kernel revision
Result: fail
Getting kernel source tree...
Source directory: /tmp/kci/linux-mainline-master-v6.2-rc8
Running job...
Checking kernel revision
Result: pass
```
It looks like the k8s runner is not working as expected so there's no KUnit results, but that's probably a different kind of issues.

**Review by JenySadadia — CHANGES_REQUESTED** [review]
As per the latest changes, please update `pending_states` in `timeout`:
```
self._pending_states = [
            state.value for state in self._api.node_states
            if state != state.DONE
        ]

```

**Comment by gctucker** [comment]
> As per the latest changes, please update `pending_states` in `timeout`:
> 
> ```
> self._pending_states = [
>             state.value for state in self._api.node_states
>             if state != state.DONE
>         ]
> ```

OK I'll do that as a follow-up PR.

**Comment by gctucker** [comment]
Ah sorry I see, it's about changing `self._db` to `self._api`.  I'm rebasing and sending an update now.

**Comment by gctucker** [comment]
OK fixed this now.  I see the error in the logs:
```
AttributeError: 'Timeout' object has no attribute '_db'
```

**Comment by gctucker** [comment]
Tested OK on staging:
```
[INFO] Listening for events... 
[INFO] Press Ctrl-C to stop.
[INFO] Time                        Commit        Node Id                   State      Result    Name
[INFO] 2023-02-17 08:08:13.426516  198538cb7232  63ef35ed5a3ec83513e79d9c  Running    -         checkout
[INFO] 2023-02-17 08:09:17.364122  ffda28d272fe  63ef35ec5a3ec83513e79d9b  Available  -         checkout
[INFO] 2023-02-17 08:09:17.525189  ffda28d272fe  63ef362d5a3ec83513e79d9d  Running    -         kver
[INFO] 2023-02-17 08:09:17.533777  ffda28d272fe  63ef362d5a3ec83513e79d9e  Running    -         kunit
[INFO] 2023-02-17 08:09:50.183437  ffda28d272fe  63ef362d5a3ec83513e79d9d  Done       Pass      kver
[INFO] 2023-02-17 08:10:49.937380  198538cb7232  63ef35ed5a3ec83513e79d9c  Available  -         checkout
[INFO] 2023-02-17 08:10:50.068060  198538cb7232  63ef368a5a3ec83513e79d9f  Running    -         kver
[INFO] 2023-02-17 08:10:50.096564  198538cb7232  63ef368a5a3ec83513e79da0  Running    -         kunit
[INFO] 2023-02-17 08:11:13.799797  198538cb7232  63ef368a5a3ec83513e79d9f  Done       Pass      kver
[INFO] 2023-02-17 08:20:13.890994  ffda28d272fe  63ef35ec5a3ec83513e79d9b  Closing    -         checkout
[INFO] 2023-02-17 08:21:14.660325  198538cb7232  63ef35ed5a3ec83513e79d9c  Closing    -         checkout
```
The timeout service didn't crash so it's all looking fine now.

**Comment by gctucker** [comment]
@JenySadadia Please take another look, I believe this is all ready to be merged now.

File: src/runner.py [hunk 1]
```diff
@@ -22,13 +23,13 @@ class Runner(Service):
 
     def __init__(self, configs, args):
         super().__init__(configs, args, 'runner')
-        self._db_config_yaml = self._db_config.to_yaml()
+        self._api_config_yaml = yaml.dump(self._api_config)
```
Comment (JenySadadia) [inline src/runner.py:26]: I would like to know why this has changed to not use `to_yaml`.

Comment (JenySadadia) [reply src/runner.py:26]: OK

**Review by JenySadadia — APPROVED** [review]

**Pull request merged**